db/fixtures/ae_datastore/Deployment/System/StateMachine/Deployment.class/__methods__/add_provider.rb

```diff
@@ -0,0 +1,65 @@
+require 'net/ssh'
+require 'rest_client'
+
+def provider_token
+  token = ""
+  Net::SSH.start($evm.root['deployment_master'], $evm.root['user'], :paranoid => false, :forward_agent => true,
+                 :key_data => $evm.root['private_key']) do |ssh|
+    cmd = "oc get -n management-infra sa/management-admin --template='{{range .secrets}}{{printf " + '"%s\n"' \
+          " .name}}{{end}}' | grep token"
+    token_key = ssh.exec!(cmd)
+    token_key.delete!("\n")
+    cmd = "oc get -n management-infra secrets #{token_key} --template='{{.data.token}}' | base64 -d"
+    token = ssh.exec!(cmd)
+  end
+  token
+end
+
+def add_provider
+  $evm.log(:info, "**************** #{$evm.root['ae_state']} ****************")
+  deployment_master = $evm.root['deployment_master']
+  token = provider_token
+
+  url = $evm.root['manageiq_url']
+  query = "/api/providers"
```

Comment:
you are right, but I'm dropping the use of api request from the automation in favor of exposed servicemodel methods and fields.